presto-spi/src/main/java/com/facebook/presto/spi/plan/TableScanNode.java

```diff
@@ -101,6 +106,15 @@ public TableHandle getTable()
         return assignments;
     }
 
+    /**
+     * Get whether this is a tableScan for deletion
+     */
+    @JsonProperty
+    public boolean isDelete()
```

Comment:
This kinda reminds me the `tempTable` field in this class early on.  Can we rather than having it inside table scan, create a new internal plan node inheriting this so that no connector can see it? @arhimondr @wenleix do you have better suggestions?

Reply:
> Previously we were implicitly exposing that the scan is a delete by rewriting the table scan table handle to reflect that the table scan is for a delete. Raptor, e.g. passes the delete transaction information in the rewritten table handle and marks rows for deletion as it does the table scan

I kinda agree with Rebecca on that. Right now a connector should plan the TableScan accordingly to support delete. So the encapsulation seems to be "kinda" broken already.

Actually do you think we can pass the delete scan plan node id in the `TableWriteInfo` instead of adding the `isDelete` field to the TableScanNode? @rschlussel How cumbersome do you think it is gonna be?

Reply:
@highker :I would say this case is a slightly different from `tempTable`. In this case, `TableScanNode` can already support delete. Thus this `delete` field itself doesn't undermine the abstraction -- it just reveals some already-broken thing under the sun. 

That being said, let's see if it's easy to hide it into `TableWriteInfo` :)